pkg/policies/controlplane/validator.go

```diff
@@ -49,11 +56,32 @@ func (v *PolicySpecValidator) ValidateSpec(
 	name string,
 	yamlSrc []byte,
 ) (bool, string, error) {
-	_, _, err := ValidateAndCompile(ctx, name, yamlSrc)
+	_, policy, err := ValidateAndCompile(ctx, name, yamlSrc)
 	if err != nil {
 		// there is no need to handle validator errors. just return validation result.
 		return false, err.Error(), nil
 	}
+
+	fluxmeters := policy.GetResources().GetFlowControl().GetFluxMeters()
+	// Deprecated: v1.5.0
+	if fluxmeters == nil {
+		fluxmeters = policy.GetResources().GetFluxMeters()
+	}
+
+	v.mu.Lock()
+	defer v.mu.Unlock()
+
+	newNames := make(map[string]bool)
+	for fluxMeterName := range fluxmeters {
+		newNames[fluxMeterName] = true
+		if v.fluxmeterNames[fluxMeterName] {
+			return false, fmt.Sprintf("fluxmeter name \"%s\" already used in other policy", fluxMeterName), nil
+		}
+	}
+	for key, val := range newNames {
+		v.fluxmeterNames[key] = val
```

Comment:
@DariaKunoichi,
We are never removing from this map. Moreover validate can be called without actually creating a Policy.

Can maintain this map inside the FluxMeter lifecycle hooks instead.

Reply:
Ok, I'll rework this.